### PR TITLE
Move check-bootstrap from a makefile rule to test::Bootstrap

### LINKED ISSUE #96688: Move `check-bootstrap` from a makefile rule to `test::Bootstrap`
It's very confusing that this can't actually be run through x.py itself; it tripped me up in https://github.com/rust-lang/rust/pull/96687.

See https://github.com/rust-lang/rust/blob/1b2e0b60cc5aec81b48835bfc322c426e2f07cd1/src/bootstrap/mk/Makefile.in#L48-L49 for the existing code and https://github.com/rust-lang/rust/blob/1b2e0b60cc5aec81b48835bfc322c426e2f07cd1/src/bootstrap/test.rs#L2349-L2355 for where it should be moved.

@rustbot label +A-rustbuild +E-easy +E-mentor +E-help-wanted +A-contributor-roadblock

### PR DESCRIPTION
**Pull request by ghost**
Fixes #96688

### CODE REVIEW

**Comment by gimbles** [comment]
(Force push was a squash)

File: src/bootstrap/mk/Makefile.in [hunk 1]
```diff
@@ -45,8 +45,6 @@ check-aux:
 		src/tools/cargo \
 		src/tools/cargotest \
 		$(BOOTSTRAP_ARGS)
-check-bootstrap:
-	$(Q)$(CFG_PYTHON) $(CFG_SRC_DIR)src/bootstrap/bootstrap_test.py
```
Comment (bjorn3) [inline src/bootstrap/mk/Makefile.in:49]: I think this should be kept except now calling the appropriate subcommand of `$(BOOTSTRAP)`.

File: src/bootstrap/test.rs [hunk 2]
```diff
@@ -2376,6 +2379,8 @@ impl Step for Bootstrap {
         // rustbuild tests are racy on directory creation so just run them one at a time.
         // Since there's not many this shouldn't be a problem.
         cmd.arg("--test-threads=1");
+
+        try_run(builder, &mut check_bootstrap);
```
Comment (bjorn3) [inline src/bootstrap/test.rs:2383]: Maybe move this to right after defining `check_bootstrap`?

File: src/bootstrap/mk/Makefile.in (hunk 1 above)
Comment (bjorn3) [reply src/bootstrap/mk/Makefile.in:49]: Never mind, it already runs as part of `./x.py test`.

File: src/bootstrap/test.rs [hunk 3]
```diff
@@ -2376,6 +2380,7 @@ impl Step for Bootstrap {
         // rustbuild tests are racy on directory creation so just run them one at a time.
         // Since there's not many this shouldn't be a problem.
         cmd.arg("--test-threads=1");
+
```
Comment (bjorn3) [inline src/bootstrap/test.rs:2383]: nit:
```suggestion
```

**Comment by gimbles** [comment]
I..have to say, you're really quick at reviewing these, I just committed 😂

**Review by jyn514 — APPROVED** [review]
Lgtm with CI passing :) thanks for working on this!

**Comment by gimbles** [comment]
image
image


There is no trailing whitespace though, just cloned the repo and ran `x fmt`, git didn't pick up any diff. 🤔

**Comment by Mark-Simulacrum** [comment]
r=me with commits squashed

**Comment by gimbles** [comment]
@Mark-Simulacrum Squashed 👍🏻

**Pull request merged**